services/common/src/main/java/com/linkedin/openhouse/common/stats/model/PartitionStats.java

```diff
@@ -0,0 +1,187 @@
+package com.linkedin.openhouse.common.stats.model;
+
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * Data model for openhouseDatasetPartitionsStats table.
+ *
+ * <p>Stores partition-level metadata and statistics such as null count, NaN count, row count, and
+ * min/max values. Can represent both partition-level and table-level statistics.
+ *
+ * <p><b>Cardinality</b>: Each partition stats record references the latest commit that modified the
+ * partition via commitMetadata.commitId (Foreign Key to {@link CommitEvent}).
+ *
+ * <p>Stats are updated/replaced when new commits modify the partition.
+ *
+ * @see CommitEvent
+ * @see CommitEventPartitions
+ * @see CommitMetadata
+ */
+@Data
+@SuperBuilder
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+public class PartitionStats extends BaseEventModels.BaseDataset {
+
+  /**
+   * Commit metadata for the latest commit that modified this partition.
+   *
+   * <p>The commitId within this metadata serves as a Foreign Key to {@link CommitEvent}.
+   */
+  @NonNull private CommitMetadata commitMetadata;
```

Comment:
A partition could be updated from multiple commits, shouldn't this be a list of commits?